lib/apiservers/engine/backends/container.go

```diff
@@ -1315,6 +1316,8 @@ func (c *Container) ContainerChanges(name string) ([]docker.Change, error) {
 			change.Kind = docker.ChangeAdd
 		case "D":
 			change.Kind = docker.ChangeDelete
+			p := strings.TrimPrefix(filepath.Base(change.Path), docker.WhiteoutPrefix)
+			change.Path = filepath.Join(filepath.Dir(change.Path), p)
```

Comment:
will change.Path ever end with a `/`? I have found that Dir("/path/dir/") ---> "/path/dir" and "". where as Dir("/path/dir/") ---> "/path" and "dir".